alhimick-prog/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+INPUT_PROMPT = 'Enter city name:'
+ERROR_BLANK_STRING = 'City name can not be blank!'
+ERROR_FILE_READ_FAIL = 'Error: data file not found'
+ERROR_MOUNT_NUMBER = 'Error: something went wrong when counting months.'
+MEAN = 'Rainfall mean:'
+VARIANCE = 'Rainfall variance:'
+EXIT_STRING = 'exit!'
+TOWN_ERROR = -1
+MOUNTH_NUMBER = 12
+CURRENT_FILE = 'data.txt'
+
+def file_reader(filename)
+  if File.exist?(filename)
+    file = File.new(filename, 'r')
+    file.sysread(file.size)
+    # I know i could use a method .readlines to get an array from file,
+    # but in the condition of assignment
+    # data is passed to methods 'mean' and 'variance' as a string.
```

Comment:
@alhimick-prog Why do you decide to send data to `mean` and `variance` as a string? 
The purpose of these methods is just to calculate `mean` and `variance`, you need to pass the data and don't think about how to process strings

Reply:
@lincAlinc "function: `mean(town, strng)` should return the average of rainfall for the city `town` and the `strng` `data `or `data1`"
On the site with the task, `data` and `data1` are strings. So in my solution the data is also passed as a string.

Reply:
@alhimick-prog ok